### PR TITLE
Audit issue 18

### PR DESCRIPTION
**Pull request by borgbob**
I might be pushing my luck here, but it seems  skipping saving when `weight == 0` would work because `query_voting_power_at_height` will return 0 even if the address is missing from storage.

### CODE REVIEW

File: contracts/cw4-voting/src/contract.rs [hunk 1]
```diff
@@ -111,12 +113,16 @@ pub fn execute_member_changed_hook(
         } else {
             negative_difference += Uint128::from(old - weight);
         }
-        USER_WEIGHTS.save(
-            deps.storage,
-            &user_address,
-            &Uint128::from(weight),
-            env.block.height,
-        )?;
+        if weight == 0 {
```
Comment (ezekiiel) [inline contracts/cw4-voting/src/contract.rs:116]: Could we add an explanatory comment here about why this works? Might be nice to add a similar one to `query_voting_power_at_height` just to make sure someone doesn't unknowingly remove the `unwrap_or_default` there.

**Review by ezekiiel — COMMENTED** [review]
This is great! Thanks. :)

Could we add a test for instantiation with zero weights and for updates that cause zero weights to be sent to the hook before we land this?

**Review by ezekiiel — APPROVED** [review]
Great work! Thanks for putting together the test.

@Callum-A would you mind giving the final stamp / merge here when you have a chance. Know you're more familiar with this code than I am.

**Review by Callum-A — APPROVED** [review]
